testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,494 @@
-        assert_equal(f.tell(), 5)
+        assert f.tell() == 5
+        assert len(f) == 10
+        assert f.tell() == 5
         f.seek(0)
-        assert_equal(f.tell(), 0)
-        ones = np.ones(30).reshape(10, 3)
-        box_compare = np.eye(3) * 20
-        for i, frame in enumerate(f):
-            xyz, box, step, time, prec = frame
-            assert_array_almost_equal(xyz, ones * i, decimal=3)
-            assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
-
-
-@run_in_tempdir()
-def test_write_xtc():
-    with XTCFile(XTC_multi_frame) as f_in, XTCFile('foo.xtc', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        for frame in f_in:
-            f_out.write(*frame)
-
-    with XTCFile('foo.xtc') as f:
-        assert_equal(len(f), 10)
+        assert f.tell() == 0
         ones = np.ones(30).reshape(10, 3)
         box_compare = np.eye(3) * 20
         for i, frame in enumerate(f):
             xyz, box, step, time, prec = frame
             assert_array_almost_equal(xyz, ones * i, decimal=3)
             assert_array_almost_equal(box, box_compare, decimal=3)
-            assert_equal(step, i)
-            assert_equal(time, i * 0.5)
-            assert_equal(prec, 1000.0)
-
-
-@run_in_tempdir()
-def test_different_box_xtc():
-    """test if we can write different box-sizes for different frames.
-    """
-    fname = 'foo.xtc'
-    orig_box = None
-    with XTCFile(XTC_multi_frame) as f_in, XTCFile(fname, 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        frame = f_in.read()
-        f_out.write(frame.x, frame.box, frame.step,
-                    frame.time, frame.prec)
-        orig_box = frame.box.copy()
-        box = frame.box.copy() + 1
-        f_out.write(frame.x, box, frame.step,
-                    frame.time, frame.prec)
-
-    with XTCFile(fname) as xtc:
-        assert_equal(len(xtc), 2)
-        frame_1 = xtc.read()
-        frame_2 = xtc.read()
-
-        assert_array_almost_equal(frame_1.box, orig_box)
-        assert_array_almost_equal(frame_1.box + 1, frame_2.box)
-
-
-@raises(IOError)
-@run_in_tempdir()
-def test_write_different_x_xtc():
-    with XTCFile(XTC_multi_frame) as f_in, XTCFile('foo.xtc', 'w') as f_out:
-        assert_equal(f_out.n_atoms, 0)
-        frame = f_in.read()
-        f_out.write(frame.x, frame.box, frame.step,
-                    frame.time, frame.prec)
-        x = np.ones((f_in.n_atoms - 1, 3))
-        f_out.write(x, frame.box, frame.step,
-                    frame.time, frame.prec)
-
-
-@raises(IOError)
-@run_in_tempdir()
-def test_write_different_prec():
+            assert step == i
+            assert time == i * 0.5
+            assert prec == 1000.0
+
+@pytest.mark.parametrize('dtype', (np.int32, np.int64, np.float32, np.float64, int, float))
+def test_write_xtc_dtype(tmpdir, dtype):
+    with tmpdir.as_cwd():
```

Comment:
So maybe having a unused fixture isn't any better, but I prefer the decorator approach to tmpdir.  We're only using the inside of the context manager, so it's cleaner to not have the indentation.